metrics/readers/metrics.go

```diff
@@ -0,0 +1,68 @@
+package readers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/tattler/data"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	api "go.opentelemetry.io/otel/metric"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+const (
+	subsystem       = "tattler"
+	eventTypeLabel  = "event_type"
+	sourceTypeLabel = "source_type"
+	changeTypeLabel = "change_type"
+	objectTypeLabel = "object_type"
+)
+
+var (
+	watchEventCount metric.Float64Counter
+	dataEntryCount  metric.Float64Counter
+)
+
+func metricName(name string) string {
+	return fmt.Sprintf("%s_%s", subsystem, name)
+}
+
+// NewRegistry creates a new Registry with initialized prometheus counter definitions
+func Init(meter api.Meter) error {
+	var err error
+	watchEventCount, err = meter.Float64Counter(metricName("watch_event_total"), api.WithDescription("total number of batches emitted by tattler"))
+	if err != nil {
+		return err
+	}
+	dataEntryCount, err = meter.Float64Counter(metricName("data_entry_total"), api.WithDescription("total number of batches emitted by tattler"))
+	if err != nil {
+		return err
+	}
+	// get object age?
```

Comment:
I shouldn't have said object age here, I wanted to know event age. It seemed possibly expensive to store the age of the event if we didn't care that much.

Reply:
It would require tracking a bunch of extra state per watch event, either by wrapping the watch event or storing it in a separate structure only used for metrics.